mesonbuild/backend/ninjabackend.py

```diff
@@ -232,6 +232,9 @@ def get_generated_headers(self, target):
             for src in genlist.get_outputs():
                 if self.environment.is_header(src):
                     header_deps.append(self.get_target_generated_dir(target, genlist, src))
+        if 'vala' in target.compilers and not isinstance(target, build.Executable):
+            hname = os.path.join('.', self.get_target_dir(target), target.vala_header)
```

Comment:
The only issue I have here is that using `.` is not consistent with the rest of the paths in the resulting `build.ninja`.